File: FLM_SubconLabelSystem/Library/Library.Database/DAL/HouseKeep.cs
using System;
using System.Data;
using System.Data.SqlClient;

namespace Library.Database.DAL
{
    public class HouseKeep : Library.SQLServer.Connection
    {
        public HouseKeep() : base("PFR_Label_DB") { }

        internal DataTable GetSubSlitChild(string Company, string datePurge, string purgeTable)
        {
            DataTable result = new DataTable();

            if (purgeTable == "SSCHILD")
                base._cmd.CommandText = "SP_HK_GET_SUB_CHILD";
            else if (purgeTable == "SSMOTHER")
                base._cmd.CommandText = "SP_HK_GET_SUB_MOTHER";
            else if (purgeTable == "SSMAIN")
                base._cmd.CommandText = "SP_HK_GET_SUB_MAIN";
            else if (purgeTable == "LOTSLIT")
                base._cmd.CommandText = "SP_HK_GET_LOT_SLIT";
            else if (purgeTable == "PC2LOT")
                base._cmd.CommandText = "SP_HK_GET_PC2_LOT";

            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;
            base._cmd.Parameters.Clear();
            base._cmd.Parameters.Add(new SqlParameter("@pCompany", Company)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@pDatePurge", datePurge)).Direction = ParameterDirection.Input;

            base._rdr = base._cmd.ExecuteReader();
            result.Load(base._rdr);
            return result;
        }

        internal string DelSubSlitChild(string pid, string pHKTable)
        {
            string result = "1";
            try
            {
                base._cmd.CommandText = "SP_HOUSEKEEPING";
                base._cmd.CommandType = CommandType.StoredProcedure;
                base._cmd.CommandTimeout = 0;
                base._cmd.Parameters.Clear();
                base._cmd.Parameters.Add(new SqlParameter("@pID", pid)).Direction = ParameterDirection.Input;
                base._cmd.Parameters.Add(new SqlParameter("@HouseKeepTable", pHKTable)).Direction = ParameterDirection.Input;

                base._cmd.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                result = ex.Message;
            }
            return result;
        }
    }
}

File: FLM_SubconLabelSystem/Library/Library.Database/DAL/HouseKeep.vb
Imports System.Data.SqlClient

Namespace DAL
    ''' <summary>
    ''' CapexCapital Data Access Layer
    ''' ------------------------------------------------
    ''' 15 March 2012  C.C.Yeon Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class HouseKeep
        Inherits Library.SQLServer.Connection

        Public Sub New()
            MyBase.New("PFR_Label_DB")
        End Sub

        Friend Function GetSubSlitChild(ByVal Company As String, ByVal datePurge As String, ByVal purgeTable As String) As DataTable
            GetSubSlitChild = New Data.DataTable

            With MyBase._cmd
                If purgeTable = "SSCHILD" Then
                    .CommandText = "SP_HK_GET_SUB_CHILD"
                ElseIf purgeTable = "SSMOTHER" Then
                    .CommandText = "SP_HK_GET_SUB_MOTHER"
                ElseIf purgeTable = "SSMAIN" Then
                    .CommandText = "SP_HK_GET_SUB_MAIN"
                ElseIf purgeTable = "LOTSLIT" Then
                    .CommandText = "SP_HK_GET_LOT_SLIT"
                ElseIf purgeTable = "PC2LOT" Then
                    .CommandText = "SP_HK_GET_PC2_LOT"
                End If
                .CommandType = CommandType.StoredProcedure
                .CommandTimeout = 0

                .Parameters.Clear()
                .Parameters.Add(New SqlParameter("@pCompany", Company)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@pDatePurge", datePurge)).Direction = Data.ParameterDirection.Input
                '.Parameters.Add(New SqlParameter("SREFData", Oracle.DataAccess.Client.OracleDbType.RefCursor)).Direction = ParameterDirection.Output

            End With

            MyBase._rdr = MyBase._cmd.ExecuteReader
            GetSubSlitChild.Load(_rdr)
        End Function
        Friend Function DelSubSlitChild(ByVal pid As String, ByVal pHKTable As String) As String
            DelSubSlitChild = "1"
            Try
                With MyBase._cmd
                    .CommandText = "SP_HOUSEKEEPING"
                    .CommandType = CommandType.StoredProcedure
                    .CommandTimeout = 0

                    .Parameters.Clear()
                    .Parameters.Add(New SqlParameter("@pID", pid)).Direction = Data.ParameterDirection.Input
                    .Parameters.Add(New SqlParameter("@HouseKeepTable", pHKTable)).Direction = Data.ParameterDirection.Input
                    '.Parameters.Add(New SqlParameter("@pCreatedCC", UpdatedCC)).Direction = Data.ParameterDirection.Input
                End With
                MyBase._cmd.ExecuteNonQuery()

            Catch ex As Exception
                DelSubSlitChild = ex.Message
            End Try
        End Function


    End Class
End Namespace